config/src/config/mod.rs

```diff
@@ -298,14 +272,26 @@ impl NodeConfig {
     }
 
     fn random_internal(&mut self, rng: &mut StdRng) {
+        let mut test = TestConfig::new_with_temp_dir();
+
         if self.base.role == RoleType::Validator {
+            test.random(rng);
+            let peer_id = PeerId::from_public_key(
+                test.account_keypair
+                    .as_ref()
+                    .unwrap()
+                    .public()
+                    .as_ref()
+                    .unwrap(),
```

Comment:
This may be able to be simplified with the better public() function i suggested. You also can use map to avoid unwrapping multiple times.

Reply:
This is fixed in the next stack when I migrate to a new KeyPair API.